src/utils/runSortReferences.ts

```diff
@@ -175,18 +175,24 @@ const runSortReferences = () => {
         return node.substring(attr.length + 2).trim();
       };
 
-      const menuItemCallback = (
+      const menuItemCallback = async (
         sortContainer: Element,
         sortBy: (
           a: { title: string; time: number },
           b: { title: string; time: number }
         ) => number
       ) => {
-        const pageTitle = getPageTitleValueByHtmlElement(sortContainer);
-        if (!pageTitle) {
+        const sidebarWindow = sortContainer.closest(".rm-sidebar-window");
+        const currentPageUID = await window.roamAlphaAPI.ui.mainWindow.getOpenPageOrBlockUid();
+        const pageUID = !!sidebarWindow
+          ? getPageUidByPageTitle(sidebarWindow.querySelector("div > a").textContent)
```

Comment:
I think the "Check if sidebar" is sufficient. Not much else we could do here besides DOM crawling.

"Find pageUID if sidebar" could be more deterministic. we could query select all of the `.rm-sidebar-window` and see what index in that array our `sidebarWindow` is in. Then, we could use `window.roamAlphaAPI.ui.rightSidebar.getWindows()` to get the pageUid.

```typescript
const sidebarWindow = sortContainer.closest(".rm-sidebar-window");
const allWindows = document.body.querySelectorAll(".rm-sidebar-window");
const sidebarWindowIndex = Array.from(allWindows).indexOf(sidebarWindow);
const sidebarWindowData = window.roamAlphaAPI.ui.rightSidebar.getWindows()[sidebarWindowIndex]
// @ts-ignore we need to ignore typescript here for now bc Roam has an API inconsistency - see my latest msg in #developers channel
const pageUID = sidebarWindowData["page-uid"] || sidebarWindowData["mentions-uid"]; 
```